pkg/api/util/util.go

```diff
@@ -8,52 +8,135 @@ package util
 
 import (
 	"crypto/subtle"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"net"
 	"net/http"
 	"strings"
 	"sync"
 
-	"github.com/DataDog/datadog-agent/pkg/api/security"
+	pkgtoken "github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/security/cert"
 	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 var (
 	tokenLock sync.RWMutex
 	token     string
 	dcaToken  string
+	// The clientTLSConfig is set by default with `InsecureSkipVerify: true`.
+	// This is intentionally done to allow the Agent to local Agent APIs when the clientTLSConfig is not yet initialized.
+	// However, this default value should be removed in the future.
+	// TODO: Monitor and fix the logs printed by GetTLSClientConfig and GetTLSServerConfig.
+	clientTLSConfig = &tls.Config{
+		InsecureSkipVerify: true,
+	}
+	serverTLSConfig *tls.Config
+	initialized     sync.Once
+	initializedBy   string
```

Comment:
Because this value only has 3 possible values (blank string, SetAuthToken, CreateAndSetAuthToken) it should be made an enum instead, backed by an int. Also, what do you think of the name `initSource`?